frontend/dynamic-demo-plugin/src/utils/telemetry.ts

```diff
@@ -0,0 +1,89 @@
+const SEGMENT_KEY = '';
+
+const initSegment = () => {
+  const analytics = ((window as any).analytics = (window as any).analytics || []);
+  if (!analytics.initialize) {
+    if (analytics.invoked)
+      window.console && console.error && console.error('Segment snippet included twice.');
+    else {
+      analytics.invoked = !0;
+      analytics.methods = [
+        'trackSubmit',
+        'trackClick',
+        'trackLink',
+        'trackForm',
+        'pageview',
+        'identify',
+        'reset',
+        'group',
+        'track',
+        'ready',
+        'alias',
+        'debug',
+        'page',
+        'once',
+        'off',
+        'on',
+        'addSourceMiddleware',
+        'addIntegrationMiddleware',
+        'setAnonymousId',
+        'addDestinationMiddleware',
+      ];
+      analytics.factory = function(e: string) {
+        return function() {
+          var t = Array.prototype.slice.call(arguments);
+          t.unshift(e);
+          analytics.push(t);
+          return analytics;
+        };
+      };
+      for (var e = 0; e < analytics.methods.length; e++) {
+        var key = analytics.methods[e];
+        analytics[key] = analytics.factory(key);
+      }
+      analytics.load = function(key: string, e: Event) {
+        const t = document.createElement('script');
+        t.type = 'text/javascript';
+        t.async = !0;
+        t.src = 'https://cdn.segment.com/analytics.js/v1/' + key + '/analytics.min.js';
+        const n = document.getElementsByTagName('script')[0];
+        if (n.parentNode) {
+          n.parentNode.insertBefore(t, n);
+        }
+        analytics._loadOptions = e;
+      };
+      analytics.SNIPPET_VERSION = '4.13.1';
+      if (SEGMENT_KEY) {
+        analytics.load(SEGMENT_KEY);
+      }
+    }
+  }
+};
+
+initSegment();
+
+export const eventListener = async (eventType: string, properties?: any) => {
+  // eslint-disable-next-line no-console
+  console.log('Demo Plugin received telemetry event: ', eventType, properties);
+  const anonymousIP = {
+    context: {
+      ip: '0.0.0.0',
+    },
+  };
+  switch (eventType) {
+    case 'identify':
+      const { user, ...otherProperties } = properties;
+      const id = user.metadata?.uid || `${location.host}-${user.metadata?.name}`;
```

Comment:
If `user` is defined, I would expect `metadata` to always be defined. So does this need to be...?

```suggestion
      const id = user?.metadata.uid || `${location.host}-${user?.metadata.name}`;
```

Reply:
I think we can omit `?` altogether then.
We can play it safe and check if user is not defined, we should not fire an identify event at all.